File: app/src/test/java/com/coinninja/coinkeeper/cn/service/CNFirebaseMessagingServiceTest.java
package com.coinninja.coinkeeper.cn.service;

import com.coinninja.coinkeeper.TestCoinKeeperApplication;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import static org.mockito.Mockito.verify;

@RunWith(RobolectricTestRunner.class)
@Config(application = TestCoinKeeperApplication.class)
public class CNFirebaseMessagingServiceTest {

    @Mock
    PushNotificationServiceManager pushNotificationServiceManager;
    private CNFirebaseMessagingService firebaseMessagingService;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        firebaseMessagingService = Robolectric.setupService(CNFirebaseMessagingService.class);

        firebaseMessagingService.pushNotificationServiceManager = pushNotificationServiceManager;
    }

    @After
    public void tearDown() throws Exception {
        pushNotificationServiceManager = null;
        firebaseMessagingService = null;
    }

    @Test
    public void updates_user_token_with_manager_when_change() {
        String token = "-- new token --";
        firebaseMessagingService.onNewToken(token);
        verify(pushNotificationServiceManager).saveToken(token);
    }

}

File: app/src/test/java/com/coinninja/coinkeeper/cn/service/CNFirebaseMessagingServiceTest.kt
package com.coinninja.coinkeeper.cn.service

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.nhaarman.mockitokotlin2.mock
import dagger.Module
import dagger.Provides
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.verify
import org.robolectric.Robolectric

@RunWith(AndroidJUnit4::class)
class CNFirebaseMessagingServiceTest {

    @Test
    fun updates_user_token_with_manager_when_change() {
        val token = "-- new token --"
        val service = Robolectric.setupService(CNFirebaseMessagingService::class.java)

        service.onNewToken(token)

        verify(service.pushNotificationServiceManager).saveToken(token)
    }

    @Module
    class CNFirebaseMessagingServiceTestModule {
        @Provides
        fun pushNotificationServiceManager(): PushNotificationServiceManager = mock()

    }
}